port/unix/omrsysinfo.c

```diff
@@ -531,6 +576,810 @@ omrsysinfo_get_CPU_architecture(struct OMRPortLibrary *portLibrary)
+	 * https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.3.0/com.ibm.zos.v2r3.iead100/CVT-map.htm */
+	uint8_t CVTFLAG4 = *(CVT + 0x17B);
+
+	return OMR_ARE_ALL_BITS_SET(CVTFLAG4, 0x02);
+}
+
+/** @internal
+ *  Check if z/OS supports the Guarded Storage Facility (GS). We use the CVTGSF (0x01) bit in the CVT structure
+ *  for the OS check.
+ *
+ *  @return TRUE if GS is supported; FALSE otherwise.
+ */
+static BOOLEAN
+getS390zOS_supportsGuardedStorageFacility(void)
+{
+	/* FLCCVT is an ADDRESS off the PSA structure
+	 * https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.3.0/com.ibm.zos.v2r3.iead300/PSA-map.htm */
+	uint8_t* CVT = (uint8_t*)(*(uint32_t*)0x10);
+
+	/* CVTFLAG3 is a BITSTRING off the CVT structure containing the CVTGSF (0x01) bit
+	 * https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.3.0/com.ibm.zos.v2r3.iead100/CVT-map.htm */
+	uint8_t CVTFLAG3 = *(CVT + 0x17A);
+
+	return OMR_ARE_ALL_BITS_SET(CVTFLAG3, 0x01);
+}
+#endif /* ifdef J9ZOS390 */
+
+/**
+ * @internal
+ * Populates J9ProcessorDesc *desc on z/OS and zLinux
+ *
+ * @param[in] desc pointer to the struct that will contain the CPU type and features.
+ *
+ * @return 0 on success, -1 on failure
+ */
+static intptr_t
+getS390Description(struct OMRPortLibrary *portLibrary, OMRProcessorDesc *desc)
+{
+/* Check hardware and OS (z/OS only) support for GS (guarded storage), RI (runtime instrumentation) and TE (transactional memory) */
+#if defined(J9ZOS390)
+#define S390_STFLE_BIT (0x80000000 >> 7)
+	/* s390 feature detection requires the store-facility-list-extended (STFLE) instruction which was introduced in z9
+	 * Location 200 is architected such that bit 7 is ON if STFLE instruction is installed */
+	if (OMR_ARE_NO_BITS_SET(*(int*) 200, S390_STFLE_BIT)) {
+		return -1;
+	}
+#elif defined(J9ZTPF)  /* defined(J9ZOS390) */
+	/*
+	 * z/TPF requires OS support for some of the Hardware Capabilities.
+	 * Setting the auxvFeatures capabilities flag directly to mimic the query_auxv call in Linux.
+	 */
+	unsigned long auxvFeatures = OMRPORT_HWCAP_S390_HIGH_GPRS|OMRPORT_S390_FEATURE_ESAN3|OMRPORT_HWCAP_S390_ZARCH|
+			OMRPORT_HWCAP_S390_STFLE|OMRPORT_HWCAP_S390_MSA|OMRPORT_HWCAP_S390_DFP|
+			OMRPORT_HWCAP_S390_LDISP|OMRPORT_HWCAP_S390_EIMM|OMRPORT_HWCAP_S390_ETF3EH;
+
+#elif defined(LINUX) /* defined(J9ZTPF) */
+	/* Some s390 features require OS support on Linux, querying auxv for AT_HWCAP bit-mask of processor capabilities. */
+	unsigned long auxvFeatures = query_auxv(AT_HWCAP);
+#endif /* defined(LINUX) */
+
+#if (defined(S390) && defined(LINUX))
+	/* OS Support of HPAGE on Linux on Z */
+	if (OMR_ARE_ALL_BITS_SET(auxvFeatures, OMRPORT_HWCAP_S390_HPAGE)){
+		setFeature(desc, OMRPORT_S390_FEATURE_HPAGE);
+	}
+#endif /* defined(S390) && defined(LINUX) */
+
+	/* Miscellaneous facility detection */
+
+	if (testSTFLE(portLibrary, 0)) {
+#if (defined(S390) && defined(LINUX))
+	/* OS Support for Linux on Z */
```

Comment:
Indentation is wrong, but I'm not convinced this comment is even needed. There are a few similar comments below, with the same issue.

Reply:
I fixed all instances of this